### ISSUE TITLE
Accumulate gradients - Gradient Clipping

### ISSUE THREAD

**Issue by manhlab** [catalyst-team/catalyst#1008]
## 🚀 Feature Request
Accumulate gradients

### Motivation
Gradient clipping may be enabled to avoid exploding gradients. Specifically, this will clip the gradient norm computed over all model parameters together.

Accumulated gradients runs K small batches of size N before doing a backwards pass. The effect is a large effective batch size of size KxN.
### Proposal
```
runner.train(
        gradient_clip= 1,
        accumulate_grad = 4
)
```

**Comment by Scitator** [comment]
Hi, already implemented in [OptimizerCallbak](https://catalyst-team.github.io/catalyst/api/callbacks.html#catalyst.callbacks.optimizer.AMPOptimizerCallback).

Example:
```python
runner.train(
  callbacks=[
    OptimizerCallbak(
      accumulation_steps=4, 
      grad_clip_params={'func': 'clip_grad_norm_', 'max_norm': 1, norm_type': 2}
    )
  ]
)
```

Nevertheless, it would be really cool if you could contribute some tutorial on Optimization tricks for our FAQ.
For example, something like [this](https://catalyst-team.github.io/catalyst/faq/logging.html), [source](https://github.com/catalyst-team/catalyst/blob/master/docs/faq/logging.rst), but `dl_trick.rst` :)

**Comment by manhlab** [comment]
thanks for that. after all found this.
